### PR TITLE
Swagger params bug fix

### PR DESCRIPTION
**Pull request by reid-andrew**
### Type of change made:
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling - no new features
### Why is this change required? What problem does it solve?
- Corrects errors in Swagger API documentation
- Displays swagger schemas properly, adds examples
- General cleanup to format of swagger documentation
### Were there any challenges that arose while implementing this feature? If so, how were they addressed?
- Learning curve around constructing swagger documentation. Lots of googling.

### CODE REVIEW

File: application/__init__.py [hunk 1]
```diff
@@ -19,7 +19,8 @@ def create_app(config_name):
         SWAGGER_URL,
         API_URL,
         config={
-            'app_name': "auctionable_change_api"
+            'app_name': "auctionable_change_api",
+            'defaultModelsExpandDepth': -1
         }
```
Comment (reid-andrew) [inline application/__init__.py:24]: This hides the "Schemas" section at the bottom of Swagger page

File: application/controllers/items.py [hunk 2]
```diff
@@ -191,6 +191,8 @@ def delete(self, item_id=None):
         item = Item.query.get(item_id)
         if not item:
             abort(404, description='That item does not exist')
+        elif not item.bids == []:
+            abort(403, description='Cannot delete item with associated bids')
         else:
```
Comment (reid-andrew) [inline application/controllers/items.py:196]: We hadn't previously accounted for this situation & API was returning a 500 error. This handles it in a more user-friendly way.

File: application/static/swagger.yml [hunk 3]
```diff
@@ -289,3 +434,147 @@ components:
...
+          type: string
+    Charities:
+      properties:
+        count:
+          type: integer
+        charities:
+          type: array
+          items:
+            $ref: '#/components/schemas/Charity'
+  examples:
```
Comment (reid-andrew) [inline application/static/swagger.yml:447]: Remainder of the file is adding data to display examples of data in requests. Previously was displaying schema instead.

File: application/static/swagger.yml [hunk 4]
```diff
@@ -174,38 +253,60 @@ paths:
...
-                $ref: '#/components/schemas/Item'
-
+                $ref: '#/components/schemas/Charities'
+              examples:
+                Bid:
+                  $ref: '#/components/examples/Charities'
         '400':
           description: Bad Request
 components:
   schemas:
```
Comment (reid-andrew) [inline application/static/swagger.yml:301]: Refactor schemas to actually reflect what we're returning via API

File: application/static/swagger.yml [hunk 5]
```diff
@@ -45,18 +48,31 @@ paths:
...
+              examples:
+                Item:
+                  $ref: '#/components/examples/Item'
         '400':
           description: Bad Request
-  /items/<id>:
+  /items/{item_id}:
     get:
       summary: Returns a selected item.
+      parameters:
```
Comment (reid-andrew) [inline application/static/swagger.yml:69]: Add in parameters so users can properly test PUT, DELETE, GET individual endpoints.

File: application/__init__.py (hunk 1 above)
Comment (kmcgrevey) [reply application/__init__.py:24]: nice deep-dive detail . . .

**Review by kmcgrevey — APPROVED** [review]
whew -- heavy detective work for a really nice bonus feature for the project.  Strong work Andy!

**Pull request merged**